### DISCUSSION TITLE
Add go to function / type start / end to `[` and `]`

### DISCUSSION THREAD

**Discussion by effinsky** [helix-editor/helix#8840] [General]
Currently we are able to do "go to previous func" and "go to next func" (applies to type and other objects as well). I miss having the option to go to end of current func, start of current func as opposed to jumping out of the current func context and going to next or prev immediately. 

This is all quite aside from the selection as being there by default -- this is congruent with basic selection-first philosophy of Helix, which I'm not debating. Bringing easier,  more comprehensive treesitter navigation for functions would be neat, though.

**Comment by pascalkuthe** [comment]
there is a command to flip selections in helix. That combined with `maf`/`mif` already covers this usecase. Yif you want a dedicated hotkey you can combined these two

**Reply by effinsky** [reply]
yeah right now I've got that covered by some remaps, but I find `]f` and `[f` are not the most common use cases. not sure you can flip selections without going into select mode, and if not, then you cannot remap this thing without actually leaving normal mode. not a dealbreaker, though hardly optimal, in my opinion.

**Reply by pascalkuthe** [reply]
* select mode is not special with some very minor exceptions you can do everything you can do in selecte mode in normal mode too. Select mode is just a different keymap but the same commands could be mapped in normal mode
* flip selection is alreayd boud to `<A-;>` in normal mode by default.
* I do think the current bindings for `]f` make sense and are consistent with all our unimpaired bindings. They are intended for quickly jummping between functions not to jump to the start of the current function. That is simply a different action covered	by different commands

**Reply by dryya** [reply]
As far as I can tell, `maf` cannot be used in a hotkey until https://github.com/helix-editor/helix/pull/14278 lands, is that right?

**Reply by MentalMegalodon** [reply]
@dryya I made it work with the following keybindings/macros:
```
A-c = "@maf<A-;>;"
A-t = "@maf;"
```
Obviously, you can replace the keys on the left with whatever you want.
